tools/test/windows/tw.cc

```diff
@@ -100,6 +100,59 @@ class TeeImpl : Tee {
   bazel::windows::AutoHandle output2_;
 };
 
+// Buffered input stream (based on a Windows HANDLE) with peek-ahead support.
+//
+// This class uses two consecutive "pages" where it buffers data from the
+// underlying HANDLE (wrapped in an AutoHandle). Both pages are always loaded
+// with data until there's no more data to read.
+//
+// The "active" page is the one where the read cursor is pointing. The other
+// page is the next one to be read once the client moves the read cursor beyond
+// the end of the active page.
+//
+// The client advances the read cursor with Move(). When the cursor reaches the
+// end of the active page, the other page becomes the active one (whose data is
+// already buffered), and the old active page is loaded with new data from the
+// underlying file.
+class IFStreamImpl : IFStream {
+ public:
+  // Creates a new IFStream.
+  //
+  // If successful, then takes ownership of the HANDLE in 'handle', and returns
+  // a new IFStream pointer. Otherwise leaves 'handle' alone and returns
+  // nullptr.
+  static IFStream* Create(bazel::windows::AutoHandle* handle,
+                          DWORD max_page_size = 0x100000 /* 1 MB */);
+
+  bool Get(uint8_t* result) const override;
```

Comment:
I think a more typical model is to advance on Read rather than having separate functions. If you want to keep separate functions, maybe call it Advance rather than Move?

Reply:
Acknowledged. I hear you, but I prefer this design.

I renamed "Move" to "Advance".

Reply:
I should've listened to you and follow the typical model.

The current model makes EOF-detection unreliable. Advance() returning false could mean an I/O error or that the read cursor was on the last byte (that Get() returned) and it cannot move forward. I have to change the current model to mimic the "typical" one.